### PR TITLE
throttle /facilities_api/v1/va to 30/m which is consistent with the average for a 2m window in datadog

### PR DESCRIPTION
**Pull request by eselkin**
## Summary

- A unique IP was issuing 27k+ requests per hour to /facilities_api/v1/va
- No other IP (search excluding the above IP) showed similar spikes
- Rate limiting the /facilities_api/v1/va to 30 requests / minute 
- Sitewide Facilities
- No flipper

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15290

## Testing done

- Tested to see if 30 reloads within one minute caused 429 on /facilities_api/v1/va
- Waited 1 minute, reload, saw 200 OK
- Test was added for this `rack_attack.rb` update in `attack_spec.rb`

## Screenshots
Screenshot 2023-09-18 at 1 37 36 PM

## What areas of the site does it impact?
* Impacts the facility locator at https://va.gov/find-locations

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (in-line documentation in code)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (datadog is monitoring 200/429)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected (no auth required)
- [x]  I added a screenshot of the developed feature

## Requested Feedback

- [ ] Test that facility locator (/find-locations) works for some location for 30 reloads (search presses or force page reload after a search has been performed) within 1 minute
- [ ] Test that facility locator blocks loading of data from vets-api after 30th load
- [ ] Test that site resumes functionality after a short pause or does not resume if repeated attempts are made.

### CODE REVIEW

**Comment by eselkin** [comment]
The code checks/linting will likely be fixed by @ryan-mcneil's PR that fixes the vulnerability that triggers the error.

File: spec/middleware/rack/attack_spec.rb [hunk 1]
```diff
@@ -119,6 +119,37 @@ def app
...
+
+      it 'limits requests' do
+        expect(last_response.status).to eq(429)
+      end
+    end
+
+    context 'response status for request from different IP' do
+      let(:other_headers) { { 'X-Real-Ip' => '4.3.2.1' } }
+
+      it 'limits requests' do
```
Comment (maxx1128) [inline spec/middleware/rack/attack_spec.rb:147]: **Nitpick:** it looks like this test is confirming the requests don't get limited, so this should probably just be changed to `'does not limit requests'`

**Review by maxx1128 — DISMISSED** [review]
I test this by going to the endpoint URL directly at `http://467558580fd610c5210b6b8fc039cb07-api.review.vetsgov-internal/facilities_api/v1/va?type=health&page=1&per_page=10&mobile=false&radius=10&lat=41.27843&long=-73.025661` and doing 30 fast refreshes. I confirmed the endpoint gave a "throttled" message, and when I went to the facility locator, the data wasn't loading there either. I went back after a minute and it was working again 👍 

Screen Shot 2023-09-19 at 12 30 02 PM
Screen Shot 2023-09-19 at 12 30 49 PM

I have a small nitpick with one of the tests, but it's not blocking.

File: spec/middleware/rack/attack_spec.rb (hunk 1 above)
Comment (eselkin) [reply spec/middleware/rack/attack_spec.rb:147]: Yep, that's true, will change that message

Comment (eselkin) [reply spec/middleware/rack/attack_spec.rb]: done

**Review by maxx1128 — APPROVED** [review]

**Pull request merged**